src/main/java/org/jabref/gui/JabRefFrame.java

```diff
@@ -1167,43 +1165,37 @@ private boolean confirmClose(LibraryTab libraryTab) {
     /**
      * Ask if the user really wants to remove any empty entries
      */
-    private Boolean confirmEmptyEntry(LibraryTab libraryTab, BibDatabaseContext context) {
+    private void confirmEmptyEntry(LibraryTab libraryTab, BibDatabaseContext context) {
         String filename = libraryTab.getBibDatabaseContext()
                                     .getDatabasePath()
                                     .map(Path::toAbsolutePath)
                                     .map(Path::toString)
                                     .orElse(Localization.lang("untitled"));
 
-        ButtonType deleteEmptyEntries = new ButtonType(Localization.lang("Delete empty entries"), ButtonBar.ButtonData.YES);
-        ButtonType keepEmptyEntries = new ButtonType(Localization.lang("Keep empty entries"), ButtonBar.ButtonData.NO);
-        ButtonType cancel = new ButtonType(Localization.lang("Return to JabRef"), ButtonBar.ButtonData.CANCEL_CLOSE);
+        if (prefs.getGeneralPreferences().shouldConfirmDeleteEmptyEntries()) {
+           boolean response = dialogService.showConfirmationDialogWithOptOutAndWait(
+                    Localization.lang("Empty entries"),
+                    Localization.lang("Library '%0' has empty entries. Do you want to delete them?", filename),
+                    Localization.lang("Delete"),
+                    Localization.lang("Cancel"),
```

Comment:
Hi @koppor , you are right.
It makes much more sense to popup a dialog asking about keeping empty entries instead of deletion.
Please keep me updated if changes needed here.